include/envoy/upstream/load_balancer.h

```diff
@@ -57,5 +57,56 @@ class LoadBalancer {
 
 typedef std::unique_ptr<LoadBalancer> LoadBalancerPtr;
 
+/**
+ * Factory for load balancers.
+ */
+class LoadBalancerFactory {
+public:
+  virtual ~LoadBalancerFactory() {}
+
+  /**
+   * @return LoadBalancerPtr a new load balancer.
+   */
+  virtual LoadBalancerPtr create() PURE;
+};
+
+typedef std::shared_ptr<LoadBalancerFactory> LoadBalancerFactorySharedPtr;
+
+/**
+ * A thread aware load balancer is a load balancer that is global to all workers on behalf of a
```

Comment:
Hmm. "GlobalLoadBalancer"? "GlobalLoadBalancerFactory"? "PerWorkerLoadBalancerFactory"? Let me know if any of those sound better and I can change.

Reply:
Yeah, can't say that I think any of these (or my suggestion) are actually much better. The intuition is that we now have a LB that is instantiated per cluster globally, and then provides per-worker instances with some shared, immutable state. I think if we were starting from scratch, you could call the "thread aware load balancer" a "GlobalLoadBalancerFactory" and the thread local LB a "ThreadLocalLoadBalancer". But, TBH, none of this will be that much more helpful to someone looking at this without reading the docs anyway.